### ISSUE TITLE
Tweak logic for showing Envoy data (xds config, stats, clusters)

### ISSUE THREAD

**Issue by kleinfreund** [kumahq/kuma-gui#1565]
### Description

Blocked on https://github.com/kumahq/kuma/issues/5216.

Once https://github.com/kumahq/kuma/issues/5216 is implemented:

- Remove the logic for only querying Envoy data when an associated resource is online (handled in https://github.com/kumahq/kuma-gui/pull/1809)
  - This logic doesn’t make sense: we would make an additional request (to load ZoneOverview) in order to potentially avoid the request for the Envoy data (meaning we actually make two requests in the positive case)
  - This logic is broken because for ZoneIngress and ZoneEgress we don’t look at the corresponding ZoneOverview’s status but the respective ZoneIngreeOverview/ZoneEgressOverview
- Make sure we correctly display the error response information based on however https://github.com/kumahq/kuma/issues/5216 will be solved

In other words, we distinguish (in the backend) between the following scenarios:

- Resource is online and we get Envoy data
- Resource is offline and we get a 400 (or something) and tell people why they can’t see Envoy data
- Resource doesn’t exist. Standard 404

**Comment by johncowen** [comment]
Just wanted to note here that we shouldn't be querying any Envoy API admin calls if the Zone is offline. From reading the description here I couldn't figure out if this issue meant we would still query the Envoy admin API if the zone was offline:

> Remove the logic for only querying Envoy data when an associated resource is online

We shouldn't query Envoy admin APIs if the zone is offline.

We could consider either:

- Showing warnings in the tabs themselves that we can't show you anything if the Zone is offline (which I think we kinda do now)
- Just don't show the tabs, and maybe showing a warning at the top of the overview saying you can see Envoy data once the zone comes online.

**Comment by kleinfreund** [comment]
> Just wanted to note here that we shouldn't be querying any Envoy API admin calls if the Zone is offline.

Why?

**Comment by slonka** [comment]
> Why?

This results in an error log on our side

```
stream to execute Stats operations is not yet connected
```

**Comment by kleinfreund** [comment]
And it’s better to avoid this error at the cost of adding a new API call in the UI (calling /zones+insights so we know whether a Zone is online)?

**Comment by slonka** [comment]
> And it’s better to avoid this error at the cost of adding a new API call in the UI (calling /zones+insights so we know whether a Zone is online)?

Don't we already have this info? If not then I guess it might be better to change the level of that log message. WDYT @lukidzi ?

**Comment by kleinfreund** [comment]
We have _access_ to it, but we currently don’t query. We’re talking about the Zone Ingress and Zone Egress views here. For those, we load ZoneIngressOverview and ZoneEgressOverview respectively, but we actually need to look at the online status of ZoneOverview (correct?). That call, we don’t currently do and it would be extra.

**Comment by johncowen** [comment]
Does the fact that we are nesting the 'gresses inside of the Zone change anything here? (see https://github.com/kumahq/kuma-gui/pull/1586)

We should probably load the ZoneOverview around all the tabs (looks like we need it for the overview and config tab anyway), and then also reuse that data in the new ingress and egress tabs.

**Comment by lahabana** [comment]
Blocked on backend as we need more information about why the info is not available

**Comment by lahabana** [comment]
We don't understand this issue anymore. It hasn't come up for a while so it's probably already fixed
